sentry_streams/sentry_streams/pipeline/pipeline.py

```diff
@@ -155,6 +166,42 @@ class Filter(TransformStep[bool]):
     step_type: StepType = StepType.FILTER
 
 
+@dataclass
+class Branch(Step):
+    """
+    A Branch represents one branch in a pipeline, which is routed to
+    by a Router.
+    The name of the Branch step must match the key of the step in
+    its Router's routing table.
+    """
+
+    step_type: StepType = StepType.BRANCH
+
+
+@dataclass
+class Router(WithInput):
+    """
+    A step which takes a routing table of Branches and sends messages
+    to those branches based on a routing function.
+    Routing functions must only return a single output branch, routing
+    to multiple branches simultaneously is not currently supported.
+    """
+
+    routing_function: Callable[..., str]
+    routing_table: Mapping[str, Branch]
+    step_type: StepType = StepType.ROUTER
+
+    def __post_init__(self) -> None:
+        super().__post_init__()
+        for branch_name in self.routing_table:
+            # ensure name of branch step matches key of routing table
+            branch_step = self.routing_table[branch_name]
+            assert (
+                branch_name == branch_step.name
+            ), f'Branch name must match its routing table key - please change "{branch_name}" to "{branch_step.name}".'
```

Comment:
Why do we need this ?

I don't think this equality should be needed.
the translator returns a mapping of step name to branch object.
You add the step name with the branch object in the `step_streams` map. The name of the Branch step is not related with the output of the routing function which is the key of the routing table.